pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -29,24 +29,47 @@ def _get_failed_tests(launch, fail_args):
     return tests
 
 
-def _get_test_collection(rp_failed_tests, items):
+def _get_skipped_tests(launch):
+    """Returns skipped test names from Report Portal Launch based on arguments"""
+    tests = launch.tests(status='skipped').keys()
+    # Formating test names for 'member of pytest test items' operation
+    tests = [str(test).replace('::', '.') for test in tests]
+    return tests
+
+
+def _get_test_collection(selectable_tests, items):
     """Returns the selected and deselected items"""
     # Select test item if its in failed tests else deselect
     LOGGER.debug('Selecting/Deselecting tests based on latest launch test results..')
     selected = []
     deselected = []
     for item in items:
         test_item = f"{item.location[0]}.{item.location[2]}"
-        if test_item in rp_failed_tests:
+        if test_item in selectable_tests:
             selected.append(item)
         else:
             deselected.append(item)
     return selected, deselected
 
 
+def _validate_launch(launch, sat_version):
+    """Stop session based on RP Launch statistics and info"""
+    if launch.info['isProcessing']:
+        raise LaunchError(f'The launch of satellite version {sat_version} is not Finished yet')
+    fail_percent = round(
+        (int(launch.statistics['failed']) / int(launch.statistics['total']) * 100)
+    )
+    if fail_percent > 20:
```

Comment:
Since 20 is just some arbitrary number that makes sense right now, I think it should be user-configurable. Overall, you don't want to change entire framework when some detail about simple business rule changes.
Also, it's not far-fetched to think about some more strict/relax pipeline, possibly running in parallel, where different number would make more sense. For that case, you want to supply different number through configuration file or environment variable.

Reply:
Make sense, For fail_percent configurable I will raise a separate PR as this PRs intent is different.

Reply:
Raised another PR for fail_percent configuraable  #8056.